Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -2173,9 +2172,27 @@ private void UpdateResourceAvailability(PerformanceCollector.MachinePerfInfo per
             }
 
 #if PLATFORM_OSX
-            Memory.PressureLevel pressureLevel = Memory.PressureLevel.Normal;
-            var result = Memory.GetMemoryPressureLevel(ref pressureLevel) == Dispatch.MACOS_INTEROP_SUCCESS;
-            resourceAvailable &= !(result && (pressureLevel > m_configuration.Schedule.MaximumAllowedMemoryPressureLevel));
+            if (!resourceAvailable)
+            {
+                Memory.PressureLevel pressureLevel = Memory.PressureLevel.Normal;
+                var result = Memory.GetMemoryPressureLevel(ref pressureLevel) == Dispatch.MACOS_INTEROP_SUCCESS;
+
+                if (result)
+                {
+                    // If the memory pressure level is not above the configured level but we've infered resources are not available earlier,
+                    // we reset the resource availability and override the decision by looking at the current pressure level only!
+                    resourceAvailable = !(pressureLevel > m_configuration.Schedule.MaximumAllowedMemoryPressureLevel);
+                }
+                else
+                {
+                    Logger.Log.UnableToGetMemoryPressureLevel(

```

Comment:
> Logger.Log.UnableToGetMemoryPressureLevel( [](start = 19, length = 43)

"resource cancelation will be skipped all together"

Does not it mean that we should set 'resourceAvailable' to true?